Sources/AuthFoundation/Network/APIResponse.swift

```diff
@@ -29,39 +29,47 @@ public struct APIResponse<T: Decodable>: Decodable {
     public let links: [Link: URL]
     
     /// Information about the current rate limit.
-    public let rateInfo: RateLimit?
+    public let rateInfo: ApiRateLimit?
     
     /// The ID for the current request.
     public let requestId: String?
+}
+
+/// Describes information related to the organization's current rate limit.
+public struct ApiRateLimit: Decodable {
+    /// The current limit.
+    public let limit: Int
     
-    /// Describes information related to the organization's current rate limit.
-    public struct RateLimit: Decodable {
-        /// The current limit.
-        public let limit: Int
-        
-        /// The rate limit remaining.
-        public let remaining: Int
+    /// The rate limit remaining.
+    public let remaining: Int
 
-        /// The time offset from UTC when the rate limit will reset, and a request may be retried.
-        public let reset: TimeInterval
+    /// The time offset from UTC when the rate limit will reset, and a request may be retried.
+    public let reset: TimeInterval
+    
+    public let delay: TimeInterval
+    
+    init?(with httpHeaders: [AnyHashable: Any]) {
+        guard let rateLimitString = httpHeaders["x-rate-limit-limit"] as? String,
+              let rateLimit = Int(rateLimitString),
+              let remainingString = httpHeaders["x-rate-limit-remaining"] as? String,
+              let remaining = Int(remainingString),
+              let resetString = httpHeaders["x-rate-limit-reset"] as? String,
+              let reset = TimeInterval(resetString)
+        else {
+            return nil
+        }
         
-        init?(with httpHeaders: [AnyHashable: Any]) {
-            guard let rateLimitString = httpHeaders["x-rate-limit-limit"] as? String,
-                  let rateLimit = Int(rateLimitString),
-                  let remainingString = httpHeaders["x-rate-limit-remaining"] as? String,
-                  let remaining = Int(remainingString),
-                  let resetString = httpHeaders["x-rate-limit-reset"] as? String,
-                  let reset = TimeInterval(resetString)
-            else {
-                return nil
-            }
-            
-            self.limit = rateLimit
-            self.remaining = remaining
-            self.reset = reset
+        self.limit = rateLimit
+        self.remaining = remaining
+        self.reset = reset
+       
+        var date: Date? = nil
+        if let dateString = httpHeaders["Date"] as? String {
+            date = httpDateFormatter.date(from: dateString)
         }
+        
+        self.delay = reset - (date ?? Date()).timeIntervalSince1970
```

Comment:
I noticed that there isn't a minDelay. I've added that in the kotlin implementation at https://github.com/okta/okta-mobile-kotlin/pull/215. I prefer both implementations to have a similar feature set. Also, I'm not too familiar with swift, but what happens in the case when either of "date" header or "x-rate-limit-reset" header is missing or invalid?

Reply:
After having a conversation with @mikenachbaur-okta . The idea is to make this as server driven as possible, hence we did not include a random minimum delay. I'm open to suggestions though

Reply:
When falling back to local date, it might be possible for ```date``` to be greater than ```reset```. In a case like that, I think it makes sense to have a minDelay. Thoughts?

Reply:
I feel that not getting a delay from the server, or being unable to calculate a proper delay, should be a error case. Instead of guessing a minimum delay, we should just fall through to the error case.

We have another ticket to provide out-of-the-box time sync/coordination support, so client clock skew shouldn't be an issue once we do that.

Reply:
From the requirements document we put together, if this is not available, we delay the min value (I think).

Reply:
Since we don't have a synchronized date support, in the unlikely case where the date is greater than the reset should we delay with the min value?

Reply:
Let's sync up about tradeoffs for including/excluding a minDelay offline. It can be added in a future PR, if that's what we decide later.

Reply:
I agree, we should discuss this further. I have a few concerns around minDelay, namely that I wouldn't want to introduce artificial delays to a user that might impact their experience.